tests/Doc/QueryEmExample.cs

```diff
@@ -0,0 +1,289 @@
+                },
+                new
+                {
+                    model = "ThrillCycle",
+                    brand = "BikeShind",
+                    price = 815,
+                    description = "An artsy,  retro-inspired bicycle that’s as functional as it is" +
+                        " pretty: The ThrillCycle steel frame offers a smooth ride. A" +
+                        " 9-speed drivetrain has enough gears for coasting in the city, but" +
+                        " we wouldn’t suggest taking it to the mountains. Fenders protect" +
+                        " you from mud, and a rear basket lets you transport groceries," +
+                        " flowers and books. The ThrillCycle comes with a limited lifetime" +
+                        " warranty, so this little guy will last you long past graduation.",
+                    condition = "refurbished",
+                },
+            };
+
+        for (var i = 0; i < bicycles.Length; i++)
+        {
+            db.JSON().Set($"bicycle:{i}", "$", bicycles[i]);
+        }
+        // HIDE_END
+
+
+        // STEP_START em1
+        SearchResult res1 = db.FT().Search(
+            "idx:bicycle",
+            new Query("@price:[270 270]")
+        );
+        Console.WriteLine(res1.TotalResults); //    >>> 1
+
+        SearchResult res2 = db.FT().Search(
+            "idx:bicycle",
+            new Query().AddFilter(
+                new Query.NumericFilter("price", 270, 270)
+            )
+        );
+        Console.WriteLine(res2.TotalResults); //    >>> 1
+        // STEP_END
+
+        // Tests for 'em1' step.
+        // REMOVE_START
+        Assert.Equal(1, res1.TotalResults);
+        Assert.Equal(1, res2.TotalResults);
+        // REMOVE_END
+
+
+        // STEP_START em2
+        SearchResult res3 = db.FT().Search(
+            "idx:bicycle",
+            new Query("@condition:{new}")
+        );
+        Console.WriteLine(res3.TotalResults);   // >>> 4
+        // STEP_END
+
+        // Tests for 'em2' step.
+        // REMOVE_START
+        Assert.Equal(4, res3.TotalResults);
+        // REMOVE_END
+
+
+        // STEP_START em3
+        Schema emailSchema = new Schema()
+            .AddTagField(new FieldName("$.email", "email"));
+
+        FTCreateParams emailParams = new FTCreateParams()
+            .AddPrefix("key:")
+            .On(IndexDataType.JSON);
+
+        db.FT().Create("idx:email", emailParams, emailSchema);
+
+        db.JSON().Set("key:1", "$", "{\"email\": \"test@redis.com\"}");
+
+        try
+        {
+            SearchResult res4 = db.FT().Search(
+                "idx:email",
+                new Query("test@redis.com").Dialect(2)
+            );
+            Console.WriteLine(res4.TotalResults);   // >>> 1
+        }
+        catch
+        {
+            Console.WriteLine("'test@redis.com' syntax not supported.");
+        }
+        // STEP_END
+
+        // Tests for 'em3' step.
+        // REMOVE_START
+
```

Comment:
It'd be helpful to check here if `res4` meets the expectations, like a number of results. Probably, should be moved inside try..catch block

Reply:
i guess this is due to dialect(2) does not handle that query.

Reply:
@uglide @atakavci The [CLI snippet](https://redis.io/docs/latest/develop/interact/search-and-query/query/exact-match/#tag-field) in the doc page shows this query working with dialect 2. It doesn't work with redis-py, but it does work with jedis if you use their [escapeQuery()](https://github.com/redis/jedis/blob/master/src/main/java/redis/clients/jedis/search/RediSearchUtil.java) method.

I couldn't get this query to work with NRedisStack, so this example is just to show that it's valid syntax but not supported yet (there's an equivalent example for redis-py, btw). If it actually is supported then I'd love to hear about how to fix the example :-)

Reply:
@andy-stark-redis jedis example employs the field name and `escapeQuery` only handles the special chars in query text.
lets stick with original example from redis_cli and employ the field name;
 something like this `@email:{"test@redis.com"}`
this will do the job.
i ll take a look later on how python and node handle it without field specifier.

Reply:
@atakavci I got this working eventually (there seems to be an issue with redis-py for this query and I thought NRedisStack had the same problem, but it actually doesn't). There's no need for the try..catch now that this is a working example, btw.